lsp-mode.el

```diff
@@ -7380,10 +7380,15 @@ This avoids overloading the server with many files when starting Emacs."
                                       (propertize "ERROR" 'face 'error)))))
                  (-partition 2 checks))))))
 
-(defun lsp-diagnose ()
+(defvar company-backends)
```

Comment:
How can just this fixing the mentioned issue?
From what I saw, the issue is when `company-mode` is **not** installed, thus forwarding declaration here will not help.
The forwarding declaration is only for byte compiling, for this issue, you may want to check for `(boundp 'company-backends)` instead.

Reply:
How before it used to work without `boundp`?

Reply:
Does it work before without `company-mode`?
Maybe user set `company-backends`, but when I test with `emacs -Q`, it failed.
For example:
``` elisp
(setq lsp-completion-provider :none)    ; so company-mode will not be activated
(package-initialize)
< open any file lsp-mode support >
M-x lsp
M-x lsp-doctor
```
it failed.

Reply:
I see, I'll open another PR changing to use `boundp`, thanks!